### ISSUE TITLE
macOS still missing some libs after build: QtDbus, libprotobuf.

### ISSUE THREAD

**Issue by fractalshine** [Mahdi-zarei/nekoray#199]
## Describe the problem

Expected behavior: successful app launch

Actual behavior: App crash due to missing libs: QtDbus, libprotobuf.

## How to reproduce

**Build environment** : 
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.7-arm64
CLT: 15.3.0.0.1.1708646388

**Testing environment**:
 macOS 15.0.1-arm64

## investigation

Not sure if my build setup configured in right way. 

**QtDbus**
To resolve problem with QtDbus changed CmakeLists.txt, added 
```find_package(Qt6 REQUIRED COMPONENTS DBus)
target_link_libraries(mytarget PRIVATE Qt6::DBus)
```
according documentation https://doc.qt.io/qt-6/qtdbus-index.html

**protobuf**

Got error with compilation process of protobuf (C++ versions less than C++14 are not supported)
resolved according to https://github.com/protocolbuffers/protobuf/issues/12393
`-DCMAKE_CXX_STANDARD=14` to cmake arguments `libs/build_deps_all.sh`

Protobuf now compiled w/o errors but crash still occurs, libprotobuf is missing

**Comment by Mahdi-zarei** [comment]
The build process should involve running `libs/build_deps_all.sh` to build the deps, then running cmake and ninja to build the app, and if you wanna ship it to another system, use `macdeployqt` to pack it up. Did you do these and get these errors in the end?

**Comment by fractalshine** [comment]
Yep did this steps as you described. On my macos build env app runs well because this libs are present in system. But on clean macos i'm experiencing crashes (libprotobuf is missing). Your builds from github actions run well on both of my macbooks.
